assets/src/edit-story/components/canvas/carousel/useCarouselKeys.js

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useMemo } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useEscapeToBlurEffect, useKeyDownEffect } from '../../keyboard';
+import { useStory } from '../../../app';
+import { duplicatePage } from '../../../elements';
+
+/**
+ * @param {Object} ref
+ * @param {Object} pageRefs
+ * @param {boolean} isRTL
+ */
+function useCarouselKeys(ref, pageRefs, isRTL) {
+  const {
+    addPageAt,
+    arrangePage,
+    currentPage,
+    currentPageId,
+    deletePage,
+    pages,
+    setCurrentPage,
+  } = useStory(
+    ({
+      state: { pages, currentPage, currentPageId },
+      actions: { addPageAt, arrangePage, deletePage, setCurrentPage },
+    }) => ({
+      addPageAt,
+      arrangePage,
+      currentPage,
+      currentPageId,
+      deletePage,
+      pages,
+      setCurrentPage,
+    })
+  );
+
+  const pageIds = useMemo(() => pages.map(({ id }) => id), [pages]);
+
+  useEscapeToBlurEffect(ref);
+
+  // Navigate left and right.
+  useKeyDownEffect(
```

Comment:
Don't we need to `stopPropagation()` here? Or is that default in the key handler?

Reply:
Yes, `stopPropagation()` is always default with Mousetrap.